src/lib/fauna.mjs

```diff
@@ -0,0 +1,31 @@
+export const getV10Client = async ({ url, secret }) => {
+  const { Client } = (await import("fauna")).default;
+  return new Client({
+    secret,
+    endpoint: url,
+  });
+};
+
+export const runV10Query = async ({
+  query,
+  url = undefined,
+  secret = undefined,
+  client = undefined,
+  options = {
+    format: "simple",
+    typecheck: false,
+  },
+}) => {
+  let _client = client;
+
+  if (!_client && url && secret) {
+    _client = await getV10Client({ url, secret });
+  } else if (!_client) {
+    throw new Error("No client provided and no url and secret provided");
```

Comment:
how about pushing the validation all the way down to `getV10Client`? there's validation at the very top of the stack (argv); might make more sense for this validation to live at the very bottom of the stack and let the middle parts assume valid input.

Reply:
Sure I can push it to `getV10Client`. What's the downside of doing validation in `runV10Query` though?

Reply:
I kept a bit of validation in `runV10Query` just to sanity check the args - let me know if you think there's a better way to do this though